src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -1,138 +1,433 @@
+  isSearchable: boolean;
   /** Boolean value used to control how the MultiSelect component will render within the page and interact with other DOM elements.
    * The default value is false. */
-  isBlockElement?: boolean;
+  isBlockElement: boolean;
   /** The items to be rendered in the Multiselect as checkbox options. */
   items: MultiSelectItem[];
+
+  listOverflow?: multiSelectListOverflowTypes;
   /** The label text rendered within the MultiSelect. */
-  labelText: string;
+  buttonText: string;
+  /** Numeric value to set the maximum number of checkbox items
+   * The default value is 5. */
+  defaultItemsVisible: number;
   /** The action to perform for clear/reset button of MultiSelect. */
   onClear?: () => void;
-  /** The type of MultiSelect that will be rendered. */
-  type: "listbox" | "dialog";
+  /** The action to perform on the checkbox's onChange function.  */
+  onChange?: MultiSelectOnChange;
+  /** The action to perform for a mixed state checkbox (parent checkbox). */
+  onMixedStateChange?: MultiSelectOnChange;
   /** The selected items state (items that were checked by user). */
   selectedItems: SelectedItems;
   /** Value used to set the width for the MultiSelect component. */
   width?: MultiSelectWidths;
-}
-type ListboxOnChange = (selectedItem: MultiSelectItem, id: string) => void;
-type DialogOnChange = (event: React.ChangeEvent<HTMLInputElement>) => void;
-
-type MultiSelectVariantsProps =
-  | {
-      type: "listbox";
-      onApply?: never;
-      /** The action to perform for downshift's onSelectedItemChange function. */
-      onChange: ListboxOnChange;
-      // These are props that are never allowed on the listbox type.
-      onMixedStateChange?: never;
-    }
-  | {
-      type: "dialog";
-      /** The action to perform for save/apply button of multiselect. */
-      onApply: () => void;
-      /** The action to perform on the checkbox's onChange function.  */
-      onChange: DialogOnChange;
-      /** The action to perform for a mixed state checkbox (parent checkbox). */
-      onMixedStateChange?: DialogOnChange;
-    };
-
-export type MultiSelectProps = MultiSelectCommonProps &
-  MultiSelectVariantsProps;
+};
 
 /**
  * The `MultiSelect` component is a form input element that presents a list
  * of `Checkbox` components from which a user can make one or multiple
- * selections. Two variants of the MultiSelect component are offered, each with
- * slightly different functionality and requirements.  Because of these
- * differences, the two variants are broken out in separate stories below.
+ * selections. The MultiSelect allows for an optional set of child checkboxes to be passed,
+ * which makes the “parent” checkbox function as a check/uncheck all toggle.
+ * If all of the children checkboxes are checked, the parent isChecked prop will be true.
+ * If only some of the child checkboxes are checked, the parent checkbox will have a isIndeterminate state,
+ * that can be checked or unchecked.
```

Comment:
This description comes up twice in the doc, once in the beginning and then again in the "With Children Items Example". I think you can remove this text here and add  a more general description of the component.